pkg/controller/deployment/deployment_controller.go

```diff
@@ -22,79 +22,314 @@ import (
 	"time"
 
 	"github.com/golang/glog"
+
 	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/client/cache"
 	"k8s.io/kubernetes/pkg/client/record"
 	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/controller"
+	"k8s.io/kubernetes/pkg/controller/framework"
 	"k8s.io/kubernetes/pkg/fields"
 	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/runtime"
 	"k8s.io/kubernetes/pkg/util"
 	deploymentutil "k8s.io/kubernetes/pkg/util/deployment"
+	"k8s.io/kubernetes/pkg/util/workqueue"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+const (
+	// FullDeploymentResyncPeriod means we'll attempt to recompute the required replicas
+	// of all deployments that have fulfilled their expectations at least this often.
+	// This recomputation happens based on contents in the local caches.
+	FullDeploymentResyncPeriod = 30 * time.Second
```

Comment:
It's 12h by default - in (non-scalability) e2e tests we are using something much smaller to test the "resync" path (I think it is 5 minutes). But yes - it would be great if you could change it would be great. You can take a look here:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/app/controllermanager.go#L267
and see how we are using ResyncPeriod function.